### PR TITLE
Bicep v0.18 support

### PR DESCRIPTION
**Pull request by PalmEmanuel**
- Implements support for Bicep assemblies version 0.18 about to be released.
- Bumps the minimum PowerShell version to 7.3 for the PS module because of .NET 7.

### CODE REVIEW

File: BicepNet.PS/LoadContext/DependencyAssemblyLoadContext.cs [hunk 1]
```diff
@@ -30,8 +33,10 @@ protected override Assembly Load(AssemblyName assemblyName)
         string assemblyFileName = $"{assemblyName.Name}.dll";
 
         // Make sure we allow other common PowerShell dependencies to be loaded by PowerShell
-        // But specifically exclude Newtonsoft.Json since we want to use a different version here
-        if (!assemblyName.Name.Equals("Newtonsoft.Json", StringComparison.OrdinalIgnoreCase))
+        // But specifically exclude certain assemblies like Newtonsoft.Json and System.Text.Json since we want to use different versions here for Bicep
+        if (!assemblyName.Name.Equals("Newtonsoft.Json", StringComparison.OrdinalIgnoreCase) &&
+            !assemblyName.Name.Equals("System.Text.Json", StringComparison.OrdinalIgnoreCase) &&
+            !assemblyName.Name.Equals("System.Text.Encodings.Web", StringComparison.OrdinalIgnoreCase))
```
Comment (PalmEmanuel) [inline BicepNet.PS/LoadContext/DependencyAssemblyLoadContext.cs:39]: This isn't pretty, we can improve it later with a list to check against conflicting assemblies with PowerShell that we want to load into ALC.

**Comment by PalmEmanuel** [comment]
Works (on my machine, after walking through dependency hell), ready for review!

File: BicepNet.PS/LoadContext/DependencyAssemblyLoadContext.cs (hunk 1 above)
Comment (SimonWahlin) [reply BicepNet.PS/LoadContext/DependencyAssemblyLoadContext.cs:39]: It will do fine for now :)

**Review by SimonWahlin — APPROVED** [review]
Looking good!

**Pull request merged**